alf/utils/summary_utils.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tensorboard.compat import tf2 as tf
+from tensorboard.plugins.histogram import metadata
+
+
+def histogram_discrete(name, data, buckets, step=None, description=None):
+    """histogram for discrete distributions
+
+    Args:
+        name (str): name for this summary
+        data (Tensor): A `Tensor` of any shape. Must be castable to `float64`
+        buckets (int): The output will have this many buckets
+        step (None|tf.Variable):  step value for this summary. this defaults to
+            `tf.summary.experimental.get_step()`
+        description (str): Optional long-form description for this summary
+    """
+    summary_metadata = metadata.create_summary_metadata(
+        display_name=None, description=description)
+    summary_scope = (getattr(tf.summary.experimental, 'summary_scope', None)
+                     or tf.summary.summary_scope)
+    with summary_scope(
+            name, 'histogram_summary', values=[data, buckets, step]) as (tag,
+                                                                         _):
+        with tf.name_scope('buckets'):
+            bucket_count = buckets
+            one_hots = tf.one_hot(
+                tf.reshape(data, shape=[-1]), depth=bucket_count)
+            bucket_counts = tf.cast(
+                tf.reduce_sum(input_tensor=one_hots, axis=0), tf.float64)
+            edge = tf.cast(tf.range(bucket_count), tf.float64)
```

Comment:
should  action_spec.maximum and action_spec.minimum be taken into account for creating edge?

Reply:
action_spec.minimum is zero for discrete actions 

https://github.com/tensorflow/agents/blob/d7905cfc8b317900657113c34f51244ec5211b69/tf_agents/environments/gym_wrapper.py#L60-L80